buildSrc/src/main/java/quilt/internal/tasks/lint/SpellingChecker.java

```diff
@@ -0,0 +1,109 @@
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class SpellingChecker implements Checker<Entry<?>> {
+    private static final String PACKAGE_INFO_CLASS_PACKAGE = "net/minecraft/unused/packageinfo/";
+
+    private static final Set<String> ALLOWED_WORDS = new HashSet<>();
+    static {
+        // collect allowed words
+        URL url;
+        try {
+            url = new URL("https://raw.githubusercontent.com/ix0rai/qm-base-allowed-wordlist/main/allowed_english_words.txt");
+            Set<String> englishWords = new java.util.Scanner(url.openStream()).useDelimiter("\\A").next().lines().collect(Collectors.toSet());
+            ALLOWED_WORDS.addAll(englishWords);
+        } catch (Exception e) {
+            throw new RuntimeException("failed to download and read english word file for spell check in mapping lint task!");
+        }
+
+        // collect minecraft words
+        InputStream parsedFile = SpellingChecker.class.getClassLoader().getResourceAsStream("minecraft_specific_words.txt");
+        if (parsedFile == null) {
+            throw new RuntimeException("could not find minecraft word file for spelling check in mapping lint task!");
+        }
+
+        Set<String> minecraftWords = getLines(parsedFile);
+        // ignore comments and empty lines
+        Set<String> minecraftWordsCopy = Set.copyOf(minecraftWords);
+        for (String word : minecraftWordsCopy) {
+            if (word.isEmpty() || word.startsWith("//")) {
+                minecraftWords.remove(word);
+            }
+        }
+
+        ALLOWED_WORDS.addAll(minecraftWords);
+    }
+
+    private static Set<String> getLines(InputStream stream) {
+        return new java.util.Scanner(stream).useDelimiter("\\A").next().lines().collect(Collectors.toSet());
+    }
+
+    @Override
+    public void check(Entry<?> entry, EntryMapping mapping, Function<Entry<?>, AccessFlags> accessProvider, ErrorReporter errorReporter) {
+        String name = mapping.targetName();
+
+        if (name != null
+                && !name.startsWith(PACKAGE_INFO_CLASS_PACKAGE)
+                // ignore unmapped methods and classes
+                && !(entry instanceof MethodEntry && name.startsWith("m_"))
+                && !(entry instanceof ClassEntry && name.startsWith("C_"))) {
+            Set<String> words = new HashSet<>();
+
+            // fixme optimise!
+            // split by slashes
+            String[] slashSeparatedWords = name.split("/");
+            for (String word : slashSeparatedWords) {
+                // split by underscores
+                String[] underscoreSeparatedWords = word.split("_");
+                for (String underscoreSeparatedWord : underscoreSeparatedWords) {
+                    // split by uppercase characters and preserve
+                    words.addAll(List.of(Arrays.stream(underscoreSeparatedWord.split("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])")).map(String::toLowerCase).toArray(String[]::new)));
```

Comment:
```suggestion
                    words.addAll(Arrays.stream(underscoreSeparatedWord.split("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])")).map(String::toLowerCase).toList());
```

I'd put the `word.split(<regex>)` in another variable. The regex here could also split on digits so you don't have to handle them later

Reply:
will look into it!

Reply:
ah, now I remember why I handled digits differently
if we include them in this regex, it preserves them in the split strings, and we still have to filter them out later